canonical/ubuntu-advantage-client
# display livepatch support

## New Notes

Fixes: #913

TODO
- [x] move cache file definition to state_files.py

## Notes

While starting this I've noticed some features to break out into separate PRs and merge beforehand. They are:

- [x] #2168 move system query functions from `util.py` to a new `system.py` because I'll need to add more here and util is already getting large. This split will help organize things
- [x] #2184 refactor how we get and parse the kernel version. livepatch uses `/proc/version_signature` and needs more granular version information (specifically, the hwe revision). We need to parse the version in the same way to properly query for kernel support.
- [x] #2189 add a new function to get the arch from `lscpu`. This is the value that livepatch uses and could be slightly different from what dpkg reports.
- [x] #2169 `DateDataValue` for parsing date strings into DataObjects
- [x] #2172 `DataObjectFile`. For the cache, it will be useful to have an file object that automatically serializes/parses content to/from a DataObject. I'll make this reuse-able.

From there I think I'll have the important building blocks for this feature and the rest will probably be in this PR.

## Test Steps
- run the new integration test
- start a vm and try out the UX for an unsupported vs supported kernel when attached/unattached/enabled/disabled
- check the logs to verify the api/cache/cli are used for support information appropriately (search for "livepatch.py")

## Checklist
